pkg/distributor/forwarding/forwarding.go

```diff
@@ -413,3 +565,13 @@ func (r *request) cleanup() {
 
 	wg.Done()
 }
+
+type grpcHTTPClient struct {
+	httpgrpc.HTTPClient
+	grpc_health_v1.HealthClient
+	conn *grpc.ClientConn
+}
```

Comment:
This struct is now already duplicate quite many times, it is also in:

https://github.com/grafana/mimir/blob/0cd57ef1aa957478c1806c4b51776c407c726b64/pkg/ingester/client/client.go#L33-L37
https://github.com/grafana/mimir/blob/0cdd91f4649b1679b2abfbda5a23ee7aaf334870/pkg/querier/store_gateway_client.go#L57-L61
https://github.com/grafana/mimir/blob/f47516c174003ccf95587e2568c8e921f46859ad/pkg/querier/worker/scheduler_processor.go#L240-L244
https://github.com/grafana/mimir/blob/a9de5ccbe6686b1e9215fbd7ae4cc2851586917d/pkg/ruler/client_pool.go#L89-L93
https://github.com/grafana/mimir/blob/180ff05e4344cde85850d1091e580970ba6baff1/pkg/alertmanager/alertmanager_client.go#L121-L125

Maybe we should consider de-duplicating this, but that's out of the scope of this PR